File: fix_encoding.js
const fs = require('fs');

const filePath = 'd:\\new\\WinAi_SeeuNextLift\\src\\features\\director\\components\\CrudFeatures.tsx';

try {
  const buffer = fs.readFileSync(filePath);
  
  // Try to parse it as UTF-8
  const text = buffer.toString('utf8');
  
  // check if there are replacement characters
  if (text.includes('\uFFFD')) {
    console.log("File contains invalid UTF-8 bytes (converted to \\uFFFD).");
    // Just write it back as is (it will save \uFFFD as UTF-8 replacement chars)
    // The compiler will then just see standard valid UTF-8 strings.
    fs.writeFileSync(filePath, text, 'utf8');
    console.log("Fixed by converting invalid bytes to standard UTF-8 replacement characters.");
  } else {
    // maybe it has a byte sequence that Node's utf8 decoder handles but Next.js turbopack swc parser fails on?
    // let's just write it back to ensure it is clean utf-8.
    fs.writeFileSync(filePath, buffer.toString('utf8'), 'utf8');
    console.log("Re-saved as UTF-8.");
  }
} catch (err) {
  console.error(err);
}

File: fix_encoding.py
import os

path = r"d:\new\WinAi_SeeuNextLift\src\features\director\components\CrudFeatures.tsx"
try:
    with open(path, "rb") as f:
        content = f.read()

    try:
        text = content.decode('utf-8')
        print("File is already valid UTF-8.")
    except UnicodeDecodeError as e:
        print(f"File contains invalid UTF-8 at position {e.start}.")
        context = content[max(0, e.start-20):e.start+20]
        print(f"Context (bytes): {context}")
        
        # Try tis-620
        try:
            text = content.decode('tis-620')
            print("Successfully decoded with tis-620.")
            with open(path, "w", encoding='utf-8') as f:
                f.write(text)
            print("Fixed by decoding as tis-620 and re-saving as utf-8")
        except:
            # Try cp874
            try:
                text = content.decode('cp874')
                print("Successfully decoded with cp874.")
                with open(path, "w", encoding='utf-8') as f:
                    f.write(text)
                print("Fixed by decoding as cp874 and re-saving as utf-8")
            except:
                text = content.decode('utf-8', errors='replace')
                with open(path, "w", encoding='utf-8') as f:
                    f.write(text)
                print("Fixed by replacing invalid bytes with '?'")

except Exception as e:
    print(f"Error: {e}")
